qi-lib/flow/core/deforest-templates.rkt

```diff
@@ -0,0 +1,10 @@
+#lang racket/base
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Special syntax for templates
+
+;; These bindings are used for ~literal matching to introduce implicit
+;; producer/consumer when none is explicitly given in the flow.
+(provide cstream->list list->cstream)
+(define cstream->list #'-cstream->list)
+(define list->cstream #'-list->cstream)
```

Comment:
I'm sure you're aware that these syntax-objects have no identifier binding; I
think a typical trick I've seen is to bind these to syntax-transformers that
raise syntax errors (as in
https://github.com/benknoble/frosthaven-manager/blob/b3bfb416054d612d8226af2295cfd676246c458c/gui/render.rkt#L40-L45,
though those are syntax parameters):
```
(define-syntax (cstream->list stx)
  (raise-syntax-error #f "???" stx))
```
In a similar vein, the "special" keyword `else` is a syntax transformer that
raises a syntax error.

I just recently came across [`wrong-syntax`](https://docs.racket-lang.org/reference/syntax-util.html#%28def._%28%28lib._racket%2Fsyntax..rkt%29._wrong-syntax%29%29), too.

Reply:
This syntax is not provided outside of the internal compiler code (currently fusion and cps modules) and is used as markers of the fusable sequence within the syntax patterns of make-deforest-rewrite. This is not used to provide syntax errors.